### PR TITLE
Dispatch to Redux part 3

### PR DESCRIPTION
**Pull request by joofsh**
This PR:
- moves cancel feedback modal to redux
- removes reviewForm from state
- Moves history out of state

### CODE REVIEW

File: client/app/components/TextareaField.jsx [hunk 1]
```diff
@@ -48,6 +48,6 @@ TextareaField.propTypes = {
   onChange: PropTypes.func.isRequired,
   onKeyDown: PropTypes.func,
   type: PropTypes.string,
-  validationError: PropTypes.string,
+  errorMessage: PropTypes.string,
```
Comment (joofsh) [inline client/app/components/TextareaField.jsx]: Unrelated fix. this prop was incorrectly named

File: client/app/containers/EstablishClaimPage/EstablishClaim.jsx [hunk 2]
```diff
@@ -693,38 +613,11 @@ export default class EstablishClaim extends BaseForm {
             backToDecisionReviewText={BACK_TO_DECISION_REVIEW_TEXT}
           />
         }
-
-        {cancelModalDisplay && <Modal
```
Comment (joofsh) [inline client/app/containers/EstablishClaimPage/EstablishClaim.jsx]: Moving this to separate `CancelModal` component

File: client/app/containers/EstablishClaimPage/EstablishClaim.jsx [hunk 3]
```diff
@@ -242,54 +217,8 @@ export default class EstablishClaim extends BaseForm {
     return values;
   }
 
-  handleFinishCancelTask = () => {
```
Comment (joofsh) [inline client/app/containers/EstablishClaimPage/EstablishClaim.jsx]: EstablishClaim.jsx is shrinking!

File: client/app/establishClaim/actions/cancelModal.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import ApiUtil from '../../util/ApiUtil';
+import WindowUtil from '../../util/WindowUtil';
+import * as Constants from '../constants';
+import { getCancelFeedbackErrorMessage } from '../selectors';
+
+export const handleCancelSubmit = (_dispatch, ownProps) => () => (
+  _dispatch((dispatch, getState) => {
```
Comment (joofsh) [inline client/app/establishClaim/actions/cancelModal.js]: this is a little weird, but the best way I could think of given our eslint rules. Basically the outer dispatch is the initial action being dispatched (dispatching a function) that function itself also provides `dispatch` as a param, so we end up hitting the `no-shadow` rule (a parameter being defined in the inner & outer scope)

File: client/test/karma/establishClaim/reducers/establishClaim-test.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import { expect } from 'chai';
+import * as Constants from '../../../../app/establishClaim/constants';
+import EstablishClaimReducer, { getEstablishClaimInitialState } from
+  '../../../../app/establishClaim/reducers/index';
+
+describe('EstablishClaimReducer', () => {
```
Comment (joofsh) [inline client/test/karma/establishClaim/reducers/establishClaim-test.js]: some new reducer tests. Most of the actions are really simple and not worth testing IMO (other than through feature tests)

File: client/app/establishClaim/selectors/index.js [hunk 6]
```diff
@@ -27,3 +28,18 @@ export const getStationOfJurisdiction = createSelector(
     return station;
   }
 );
+
+
+const getIsValdating = (state) => state.isValidating;
```
Comment (shanear) [inline client/app/establishClaim/selectors/index.js]: it's kind of awkward that we have to create all these functions that are basically:

`const getThing = (state) => state.thing`

no need to fix, just noting.

File: client/app/util/WindowUtil.js [hunk 7]
```diff
@@ -0,0 +1,5 @@
+export default {
+  reloadPage() {
```
Comment (shanear) [inline client/app/util/WindowUtil.js]: not_bad

File: client/app/containers/EstablishClaimPage/EstablishClaimForm.jsx [hunk 8]
```diff
@@ -142,29 +143,28 @@ EstablishClaimForm.propTypes = {
...
-const mapStateToProps = (state, ownProps) => {
-  return {
-    establishClaimForm: state.establishClaimForm,
-    stationOfJurisdiction: getStationOfJurisdiction(
-      state.specialIssues,
-      ownProps.stationKey
-    )
-  };
-};
+const mapStateToProps = (state, ownProps) => ({
```
Comment (shanear) [inline client/app/containers/EstablishClaimPage/EstablishClaimForm.jsx]: just for clarification, these are all thunk-ified now?

File: client/app/containers/EstablishClaimPage/EstablishClaim.jsx (hunk 2 above)
Comment (shanear) [inline client/app/containers/EstablishClaimPage/EstablishClaim.jsx]: 🎻

File: client/app/containers/EstablishClaimPage/EstablishClaim.jsx (hunk 3 above)
Comment (shanear) [inline client/app/containers/EstablishClaimPage/EstablishClaim.jsx]: 🎷

File: client/test/karma/establishClaim/reducers/establishClaim-test.js (hunk 5 above)
Comment (shanear) [inline client/test/karma/establishClaim/reducers/establishClaim-test.js]: 👍

File: client/app/establishClaim/components/CancelModal.jsx [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+  };
+};
+
+const mapDispatchToProps = (dispatch, ownProps) => ({
+  handleCloseCancelModal: () => {
+    dispatch({
+      type: Constants.TOGGLE_CANCEL_TASK_MODAL
+    });
+  },
+  handleCancelSubmit: () => (
```
Comment (shanear) [inline client/app/establishClaim/components/CancelModal.jsx]: this function is still a beast. We should consider breaking it up somehow.

File: client/app/establishClaim/components/CancelModal.jsx [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+    dispatch((_dispatch, getState) => {
+      _dispatch({ type: Constants.VALIDATE_CANCEL_FEEDBACK });
```
Comment (shanear) [inline client/app/establishClaim/components/CancelModal.jsx]: why do we need `dispatch` and `_dispatch`?

File: client/app/establishClaim/selectors/index.js (hunk 6 above)
Comment (joofsh) [inline client/app/establishClaim/selectors/index.js]: yea, I agree I'm not a fan either. It's part of the `reselect` API (I can't see a way around it). 

As a small upside, I like that it allows you to lazily call `mySelector(state)` but then the actual selector function explicitly takes only a couple parameters that it uses

File: client/app/containers/EstablishClaimPage/EstablishClaimForm.jsx (hunk 8 above)
Comment (joofsh) [inline client/app/containers/EstablishClaimPage/EstablishClaimForm.jsx]: `thunk` is only for dispatching async actions (it's not used here in mapping state to props). Most actions are still synchronous and are _not_ thunkified (we should avoid thunk-ing when possible).  Only 1 action in dispatch so far is thunkified (your other comment about `_dispatch`)

File: client/app/establishClaim/components/CancelModal.jsx (hunk 9 above)
Comment (joofsh) [inline client/app/establishClaim/components/CancelModal.jsx]: I definitely thought/considered that too, but any idea I had only added complexity. I think it's good to be able to look at one method and understand everything going on when you click the submit button. Open to ideas though

File: client/app/establishClaim/components/CancelModal.jsx (hunk 10 above)
Comment (joofsh) [inline client/app/establishClaim/components/CancelModal.jsx]: So this will be needed anytime with use `thunk` (and don't use `bindActionCreators`). This is the one action where instead of returning an object, we're returning a function. That function itself takes `dispatch` as a parameter. I could call the inner scope's param `dispatch` too, but then it throws an [eslint error](http://eslint.org/docs/rules/no-shadow). The rule itself is a good one, and I thought it best for now to make the distinction clear for people learning Redux/thunk (the inner and outer dispatch are different).

Down the road we'd maybe want to add `dispatch` as an exception to the eslint rule
